patches/tModLoader/Terraria/GameContent/UI/ResourceSets/PlayerStatsSnapshot.cs.patch

```diff
@@ -0,0 +1,73 @@
+--- src/TerrariaNetCore/Terraria/GameContent/UI/ResourceSets/PlayerStatsSnapshot.cs
++++ src/tModLoader/Terraria/GameContent/UI/ResourceSets/PlayerStatsSnapshot.cs
+@@ -1,3 +_,5 @@
++using Terraria.ModLoader;
++
+ namespace Terraria.GameContent.UI.ResourceSets
+ {
+ 	public struct PlayerStatsSnapshot
+@@ -5,32 +_,58 @@
+ 		public int Life;
+ 		public int LifeMax;
+ 		public int LifeFruitCount;
+-		public float LifePerSegment;
++		// Made double to prevent bugs caused by precision limits (#HealthManaAPI)
++		public double LifePerSegment => LifeMax / (double)AmountOfLifeHearts;
+ 		public int Mana;
+ 		public int ManaMax;
+-		public float ManaPerSegment;
++		// Made double to prevent bugs caused by precision limits (#HealthManaAPI)
++		public double ManaPerSegment => ManaMax / (double)AmountOfManaStars;
```

Comment:
what was wrong with `float` here? Can you give an example?

Reply:
Let's take the example I used in the Discord thread:
1) `LifeMax` is set to `290` and `AmountOfLifeHearts` is `15` (e.g. a mod adds a "negative heart" which removes 10 max life)
2) `290 / 15d = 19.333333333333332d` and `290 / 15f = 19.333334f`
3) code that then tries to get the amount of hearts from `LifeMax / LifePerHeart` for whatever reason ends up with `15.000000000000002d` -- which gets correctly truncated to `15` -- instead of `14.999999f` -- which gets incorrectly truncated to `14`

Reply:
This issue is not exclusive to `float` over double. This can happen any time you try to do `v = a / b` and then expect  `a / v == b`

```cs
for (int i = 200; i < 400; i++) { 
     double d = i / 15d;
     int k = (int)(i / d);
     if (k != 15) Console.WriteLine(i);
}
```
```
> 295/15d
19.666666666666668
> 295/(295/15d)
14.999999999999998
```

Reply:
right.  i'll revert the `float`->`double` changes